cmd/mailbox-controller/main.go

```diff
@@ -308,65 +332,65 @@ func (ctl *mbCtl) sync1(ctx context.Context, ref any) {
 	logger := klog.FromContext(ctx)
-	wsName, ok := ref.(string)
+	ref, ok := refany.(syncTargetRef)
 	if !ok {
-		logger.Error(nil, "Sync expected a string", "ref", ref, "type", fmt.Sprintf("%T", ref))
-		return false
-	}
-	wsNameParts := strings.Split(wsName, wsNameSep)
-	if len(wsNameParts) != 2 {
-		logger.Error(nil, "Sync expected a string with 2 parts", "ref", ref, "wsNameParts", wsNameParts)
+		logger.Error(nil, "Sync expected a syncTargetRef", "ref", refany, "type", fmt.Sprintf("%T", refany))
 		return false
 	}
-	cluster := logicalcluster.Name(wsNameParts[0])
-	targetName := wsNameParts[1]
+	cluster := ref.cluster
+	targetName := ref.name
 	syncTarget, err := ctl.synctargetLister.Cluster(cluster).Get(targetName)
 	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Unable to Get referenced SyncTarget", "wsNameParts", wsNameParts)
+		logger.Error(err, "Unable to Get referenced SyncTarget", "ref", ref)
 		return true
 	}
+	wsName := ref.mailboxWSName()
 	workspace, err := ctl.workspaceLister.Get(wsName)
 	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Unable to Get referenced Workspace", "wsName", wsName)
+		logger.Error(err, "Unable to Get referenced Workspace", "ref", ref, "wsName", wsName)
 		return true
 	}
 	if syncTarget == nil || syncTarget.DeletionTimestamp != nil {
 		if workspace == nil || workspace.DeletionTimestamp != nil {
-			logger.V(3).Info("Both absent or deleting, nothing to do", "wsName", wsName)
+			logger.V(3).Info("Both absent or deleting, nothing to do", "ref", ref)
 			return false
 		}
 		err := ctl.workspaceClient.Delete(ctx, wsName, metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &workspace.UID}})
 		if err == nil || k8sapierrors.IsNotFound(err) {
-			logger.V(2).Info("Deleted unwanted workspace", "wsName", wsName)
+			logger.V(2).Info("Deleted unwanted workspace", "ref", ref)
 			return false
 		}
-		logger.Error(err, "Failed to delete unwanted workspace", "wsName", wsName)
+		logger.Error(err, "Failed to delete unwanted workspace", "ref", ref)
 		return true
 	}
 	// Now we have established that the SyncTarget exists and is not being deleted
 	if workspace == nil {
 		ws := &tenancyv1alpha1.Workspace{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: wsName,
+				Annotations: map[string]string{SyncTargetNameAnnotationKey: ref.name},
+				Name:        wsName,
 			},
 			Spec: tenancyv1alpha1.WorkspaceSpec{},
 		}
 		_, err := ctl.workspaceClient.Create(ctx, ws, metav1.CreateOptions{FieldManager: "mailbox-controller"})
-		if err != nil {
-			logger.Error(err, "Failed to create workspace", "wsName", wsName)
-			return true
+		if err == nil {
+			logger.V(2).Info("Created missing workspace", "ref", ref)
+			return false
 		}
-		logger.V(2).Info("Created missing workspace", "wsName", wsName)
-		return false
-
+		if k8sapierrors.IsAlreadyExists(err) {
+			logger.V(3).Info("Missing workspace was created concurrently", "ref", ref)
```

Comment:
Is that significant to know or treat ?

Reply:
I added this log message because I saw this situation happen when testing and I did not expect it.  Frankly, I still can not explain it.

More broadly: i tried to follow the pattern of having one debug or error message on each code path.  For situations that are not really problems, it is a debug message.  Ones that signal less significant situations are at higher V levels.